pkg/operator/starter.go

```diff
@@ -59,31 +51,53 @@ func RunOperator(ctx context.Context, cc *controllercmd.ControllerContext) error
 	}
 	versionRecorder.SetVersion("operator", status.VersionForOperatorFromEnv())
 
-	kubeInformersForTargetNamespace := v1helpers.NewKubeInformersForNamespaces(kubeClient,
+	kubeInformersForTargetNamespace := v1helpers.NewKubeInformersForNamespaces(kubeConfigContainer.KubeConfig,
 		operatorclient.TargetNamespace,
 	)
 
-	configInformers := configinformers.NewSharedInformerFactory(configClient, 10*time.Minute)
+	configInformers := configinformers.NewSharedInformerFactory(kubeConfigContainer.ConfigClient, resyncInterval)
 
 	statusController := status.NewClusterOperatorStatusController(
 		"cert-manager",
 		[]configv1.ObjectReference{
 			{Resource: "namespaces", Name: operatorclient.TargetNamespace},
 		},
-		configClient.ConfigV1(),
+		kubeConfigContainer.ConfigClient.ConfigV1(),
 		configInformers.Config().V1().ClusterOperators(),
 		operatorClient,
 		versionRecorder,
 		cc.EventRecorder,
 	)
 
-	deploymentController := deployment.NewCertManagerDeploymentController(
+	certManagerControllerDeployment := deployment.NewCertManagerControllerDeployment(
+		operatorclient.TargetNamespace, operatorclient.TargetNamespace,
+		os.Getenv("OPERAND_CERT_MANAGER_IMAGE_VERSION"),
+		operatorClient,
+		kubeConfigContainer,
+		kubeInformersForTargetNamespace.InformersFor(operatorclient.TargetNamespace),
+		kubeConfigContainer.ConfigClient.ConfigV1().ClusterOperators(),
+		cc.EventRecorder,
+		versionRecorder,
+	)
+
+	certManagerCAInjectorDeployment := deployment.NewCertManagerCAInjectorDeployment(
+		operatorclient.TargetNamespace, operatorclient.TargetNamespace,
+		os.Getenv("OPERAND_CERT_MANAGER_IMAGE_VERSION"),
+		operatorClient,
+		kubeConfigContainer,
+		kubeInformersForTargetNamespace.InformersFor(operatorclient.TargetNamespace),
+		kubeConfigContainer.ConfigClient.ConfigV1().ClusterOperators(),
+		cc.EventRecorder,
+		versionRecorder,
+	)
+
+	certManagerWebhookDeployment := deployment.NewCertManagerWebhookDeployment(
```

Comment:
instead of three functions and files and passing the same values three times , create a `CertManagerDeploymentSet()` that creates these three controllers

Reply:
Done - that was a good idea!